### ISSUE TITLE
Rule proposal: Force named tuples

### ISSUE THREAD

**Issue by lgrahl** [typescript-eslint/typescript-eslint#4261]
### Description

Unnamed tuples can be long and confusing to read, e.g. if longer tuples are sent to a worker/main via `postMessage`. TIL that the tuple items can also be named. It would be great if there were a way to enforce this consistently.

### Fail

```ts
type SerializedUser = [string, number];
type Timespan = [number, number];
```

### Pass

```ts
type SerializedUser = [name: string, lastLoginAt: number];
type Timespan = [start: number, end: number];
```

**Comment by bradzacher** [comment]
This is a stylistic lint rule which bans syntax. We generally look to avoid building new rules to do this unless there is some good value that can be gained from building a rule (eg complex checks or fixers or something).

You can alert on this this by using `no-restricted-syntax`: https://typescript-eslint.io/docs/linting/troubleshooting#how-can-i-ban-specific-language-feature

```json
"no-restricted-syntax": [
  "error",
  {
    "selector": "TSTupleType > :not(TSNamedTupleMember).elementTypes",
    "message": "You must use named tuple members.",
  },
]
```
[playground example of the above config](https://typescript-eslint.io/play#ts=4.5.2&sourceType=module&code=C4TwDgpgBAyhBOBLAhgG0QLwgEwKoGcEBGKAXigG19gkA7AcwBopaBXAWwCMEBdAbgBQoSFAAqidhHxhktEuQpsuCZku7x+AoeGhwkaTDgIIATGUq1kkgFxRqdJlFTJqAGQD29RLQCCwW2q8gsLQ4pLSsmYK1Mjw-iwc6swQtNgBiUFAA&rules=N4Igdg9gtATgpgZwC4wJYGMlwCZQQTzCQEMAPEALgG0Q4YYIYQAaUBOAGzk0cpABUAyvwCuABy798YuAAIAfLIqQkACiEA5YgFscoiXACycbQCM6ASgB0nE3CJSZCFiF0IExAOZw+ATQgistoiyLIhcmA6OLJI4lxBJuYwCFYgAL4AumlAA&tsConfig=N4XyA)
